### ISSUE TITLE
Accessibility: Elements that are not an "a" or "button" are being used for for clickable UI

### ISSUE THREAD

**Issue by yousoph** [apache/superset#26190]
#### How to reproduce the bug

1. Create a dashboard 
2. In the Inspect console, look at the different button elements 

### Expected results
Clickable elements should have `<a>` or `<button>` 
https://www.w3.org/WAI/WCAG21/Understanding/keyboard
https://www.w3.org/WAI/WCAG21/Techniques/html/H91 

### Actual results

Elements such as the tab headers in dashboards, the icon to copy a link, etc don't have `<a>` or `<button>`.
If an element is clickable, then it should semantically show as a clickable item by using the proper html semantics. 
If neither an `<a>` or `<button>` can be used, then a `role=button` attribute can be applied to the element

### Environment

(please complete the following information):

- browser type and version:
- superset version: `master`
- python version: `python --version`
- node.js version: `node -v`
- any feature flags active:

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

We may also be able to check if adding role=button works

**Comment by eulloa10** [comment]
@yousoph @eschutho I can handle this one if it hasn't been claimed. Thanks for the additional notes in the project-accessibility channel.

**Comment by eulloa10** [comment]
@eschutho I submitted #26602 for this issue. Please let me know if you have any questions, comments, or suggestions.

**Comment by rusackas** [comment]
I think this was supposed to be closed by #26602, so I'll just do that here. Holler if this needs to be reopened.
